plugin/src/main/groovy/com/novoda/staticanalysis/internal/lint/LintConfigurator.groovy

```diff
@@ -46,32 +62,29 @@ class LintConfigurator implements Configurator {
         }
     }
 
-    private void configureToolTask() {
-        // evaluate violations after lint
-        def collectViolations = createCollectViolationsTask(violations)
+    private void configureCollectViolationsTask(variant = null) {
+        def taskSuffix = variant ? variant.name : ''
+        def collectViolations = createCollectViolationsTask(taskSuffix, violations).with {
+            it.dependsOn project.tasks.findByName("lint${taskSuffix.capitalize()}")
+        }
         evaluateViolations.dependsOn collectViolations
-        collectViolations.dependsOn project.tasks['lint']
     }
 
-    private CollectLintViolationsTask createCollectViolationsTask(Violations violations) {
-        project.tasks.create('collectLintViolations', CollectLintViolationsTask) { collectViolations ->
-            collectViolations.xmlReportFile = xmlOutputFile
-            collectViolations.htmlReportFile = new File(defaultOutputFolder, 'lint-results.html')
-            collectViolations.violations = violations
+    private CollectLintViolationsTask createCollectViolationsTask(String taskSuffix, Violations violations) {
+        project.tasks.create("collectLint${taskSuffix.capitalize()}Violations", CollectLintViolationsTask) { task ->
+            def reportSuffix = taskSuffix ? "-$taskSuffix" : ''
+            task.xmlReportFile = xmlOutputFileFor(reportSuffix)
+            task.htmlReportFile = new File(defaultOutputFolder, "lint-results${reportSuffix}.html")
+            task.violations = violations
         }
     }
 
-    private File getXmlOutputFile() {
-        project.android.lintOptions.xmlOutput ?: new File(defaultOutputFolder, 'lint-results.xml')
+    private File xmlOutputFileFor(reportSuffix) {
+        project.android.lintOptions.xmlOutput ?: new File(defaultOutputFolder, "lint-results${reportSuffix}.xml")
```

Comment:
does this mean that when the user sets `xmlOutput` then every variant task will output to the same report file?

Reply:
No idea. ☺ Good question. Only way to find out is to try. I don't think the documentation mentions that.

Reply:
It overrides the files. If you have xmlOuput defined, you get a single output no matter which lint task you run. So this seems to be the right way to do. But also we should do it for HTML report actually. Same happens for that too.

Reply:
that's weird behaviour. I agree that this is in line with the default reporting, but I wonder if we should provide a better way to handle that when using variant-aware lint tasks (ie: log a warning/error, or ignore that, or try to generate variant-aware reports from `xmlOutput`).

Reply:
I agree. We can talk to Android team about this. 
I don't think it is possible to do variant-aware custom `xmlOutput` with the current lintOptions in Android build tools. It is because this configuration is done only once in the configuration phase. Whatever you do overrides the file name. You can only have 1 single name for the output. If it is null, they generate file names with variant name in it.